tests/test_csc.py

```diff
@@ -56,56 +57,113 @@
     """
 )
 
+# Timeout [s].
+TIMEOUT = 1.0
+
 
 class CscTestCase(unittest.IsolatedAsyncioTestCase):
     @contextlib.asynccontextmanager
-    @mock.patch.object(attcpip.AtSimulator, "cmd_evt_connect_callback")
     async def create_at_simulator(
-        self, mock_sim: mock.AsyncMock
+        self, go_to_fault_state: bool
     ) -> typing.AsyncGenerator[None, None]:
-        async with attcpip.AtSimulator(
-            host=tcpip.LOCALHOST_IPV4, cmd_evt_port=5000, telemetry_port=6000
-        ) as self.simulator:
-            await self.simulator.cmd_evt_server.start_task
-            await self.simulator.telemetry_server.start_task
-            yield
-
-    async def test_csc(self) -> None:
         os.environ["LSST_TOPIC_SUBNAME"] = "test_attcpip"
         os.environ["LSST_SITE"] = "test"
         os.environ["LSST_DDS_PARTITION_PREFIX"] = "test"
         attcpip.AtTcpipCsc.version = "UnitTest"
 
         with mock.patch.object(
             salobj.Controller, "_assert_do_methods_present"
-        ), mock.patch.object(
-            salobj.ConfigurableCsc, "read_config_dir"
-        ), mock.patch.object(
-            salobj.topics.WriteTopic, "write"
-        ):
-            async with self.create_at_simulator():  # type: ignore
-                csc = attcpip.AtTcpipCsc(
-                    name="Test",
-                    index=0,
-                    config_schema=CONFIG_SCHEMA,
-                    config_dir=CONFIG_DIR,
-                    initial_state=salobj.State.STANDBY,
-                    simulation_mode=1,
-                )
-                assert csc is not None
-                csc.simulator = self.simulator
-                data = salobj.BaseMsgType()
-                data.configurationOverride = ""
-                assert self.simulator.simulator_state == attcpip.SimulatorState.STANDBY
+        ), mock.patch.object(attcpip.AtSimulator, "cmd_evt_connect_callback"):
+            async with attcpip.AtSimulator(
+                host=tcpip.LOCALHOST_IPV4, cmd_evt_port=5000, telemetry_port=6000
```

Comment:
Can we use the capital variables for 5000 and 6000, or put them into the configuration file?

But actually, if they are for the local simulator, I thought the OS will randomly assign the listening port numbers in the server side.

Reply:
I extracted them to constants.